builds/ngap/beslog2json.awk

```diff
@@ -1,226 +1,271 @@
-        pretty="false";
-    }
-
-    if(pretty=="true"){
-        n="\n";
-        indent="    ";
-    }
-
+# Log Fields type==request
+# 1601646465|&|2122|&|request|&|OLFS|&|0:0:0:0:0:0:0:1|&|USER_AGENT|&|92F3C71F959B56515C98A09088CA2A8E
+#    1          2        3       4           5              6               7
+# |&|-|&|1601646465304|&|18|&|HTTP-GET|&|/opendap/hyrax/data/nc/fnoc1.nc.dds|&|u|&|BES
+#    8     9             10     11              12                             13   14
+# |&|get.dds|&|dap2|&|/Users/ndp/OPeNDAP/hyrax/build/share/hyrax/data/nc/fnoc1.nc|&|u
+#      15       16                              17                                  18
+#
 # 1601642669|&|2122|&|timing|&|TimingFields
 # 1601642669|&|2122|&|verbose|&|MessageField
 # 1601642679|&|2122|&|info|&|MessageField
 # 1601642679|&|2122|&|error|&|MessageField
 # 1601642679|&|2122|&|request|&|RequestFields
+#
+#
+# You can set the control variables from the awk command line like:
+#
+#     tail -f /var/log/bes/bes.log | awk -f beslog2json.awk -v send_info=true -v send_timing=true -v prefix=""
+#
+# -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
 
+########################################################################
+#
+# Retuns the boolean state based on the passed value and the default.
+#
+# Awk doesn't have boolean types so for this work I have it use the
+# string values of "true" and "false". If setting the value from the
+# command line like:
+#    awk -v foo=true -v bar=1
+# This function will accept "true" or 1 (and only 1) as a true value.
+#
+function process_bool_value(var_val, dfault, ret_val){
+    ret_val = dfault;
+    if (length(var_val)>0) {
+        if(var_val != "true" && var_val != "1"){
+            ret_val="false";
+        }
+    }
+    return ret_val;
+}
 
 
-    # -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
-    # Log line base.
-
-    # The time that the log message was written.
-    log_line_base[1]="time";
-
-    # The PID of the beslistener that wrote the log entry.
-    log_line_base[2]="pid";
-
-    # The message associate with the log entry
-    log_line_base[3]="log_name";
-
-
-
-    # -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
-    # Request Log Fields
-
-# 1601646255|&|2122|&|request|&|OLFS|&|0:0:0:0:0:0:0:1|&|USER_AGENT|&|92F3C71F959B56515C98A09088CA2A8E|&|-|&|1601646255543|&|10|&|HTTP-GET|&|/opendap/hyrax/data/nc/nc4_strings.nc.dds|&|-|&|BES|&|get.dds|&|dap2|&|/Users/ndp/OPeNDAP/hyrax/build/share/hyrax/data/nc/nc4_strings.nc
-# 1601646465|&|2122|&|request|&|OLFS|&|0:0:0:0:0:0:0:1|&|USER_AGENT|&|92F3C71F959B56515C98A09088CA2A8E|&|-|&|1601646465304|&|18|&|HTTP-GET|&|/opendap/hyrax/data/nc/fnoc1.nc.dds|&|u|&|BES|&|get.dds|&|dap2|&|/Users/ndp/OPeNDAP/hyrax/build/share/hyrax/data/nc/fnoc1.nc|&|u
-
-    # OLFS Tag.
-    request_log_fields[4]="OLFS";
+########################################################################
+#
+# Opens a json log element with kvp for time, pid and log entry type.
```

Comment:
This needs a comment to help people understand that 'n', 'indent' and 'prefix' are global and set in BEGIN.